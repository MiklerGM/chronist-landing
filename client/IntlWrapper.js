import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';

import localeDataRU from './locales/ru.json';
// import localeDataEN from './locales/en.json';

class IntlWrapper extends React.Component {
  render() {
    return (
      <IntlProvider
        locale={this.props.locale}
        key={this.props.locale}
        messages={this.props.messages}
      >
        {this.props.children}
      </IntlProvider>
    );
  }
}

IntlWrapper.propTypes = {
  locale: PropTypes.string,
  messages: PropTypes.object
};

IntlWrapper.defaultProps = {
  locale: 'ru',
  messages: localeDataRU
};

export default IntlWrapper;
