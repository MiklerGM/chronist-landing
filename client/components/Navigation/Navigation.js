import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import LocaleWidget from '../Footer/LocaleWidget';

import './Navigation.less';

class Navigation extends React.Component {
  state = {
    isOpen: false,
  }

  toggle(e) {
    e.preventDefault();
    this.setState(state => ({ isOpen: !state.isOpen }));
  }

  render() {
    return (
      <div>
        <div className="container-fluid">
          <nav>
            <ul className={this.state.isOpen ? 'hidden' : ''}>
              <li>
                <NavLink exact activeClassName='active' to="/">
                  <FormattedMessage id='nav.main' />
                </NavLink>
              </li>
              <li>
                <a href='https://maps.chronist.ru/'>
                  <FormattedMessage id='nav.map' />
                </a>
              </li>
              <li>
                <LocaleWidget
                  onChangeLanguage={this.props.onChangeLanguage}
                  min={true}
                  locale={this.props.locale}
                />
              </li>
              <li>
                <NavLink activeClassName='active' to="/blog">
                  <FormattedMessage
                    id='nav.blog'
                  />
                </NavLink>
              </li>
              <li>
                <a href="https://vk.com/chronist"> VK </a>
              </li>
              <li>
                <NavLink activeClassName='active' to="/faq">FAQ</NavLink>
              </li>
            </ul>
            <button className='icon' onClick={e => this.toggle(e)}>
              <i className='fa icon-menu' />
            </button>
          </nav>
        </div>
      </div>
    );
  }
}

Navigation.propTypes = {
  locale: PropTypes.string.isRequired,
  onChangeLanguage: PropTypes.func.isRequired
};

export default Navigation;
