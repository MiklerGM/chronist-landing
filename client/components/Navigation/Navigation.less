@import "../../styles/colors.less";

nav {
  background-color: transparent;
  margin: 0;
  list-style-type: none;
  overflow: hidden;
  padding: 1rem;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  & ul {
    padding: 0;
    list-style-type: none;
    display: block;
    &.hidden li:not(:first-child) {
      display: none;
    }
  }
  & a {
    font-size: 1rem;
    display: inline-block;
    cursor: pointer;
    color: #9d9d9d;
    text-align: center;
    padding: 1rem;
    text-decoration: none;
    transition: 300ms;
    box-shadow: none;
    &:hover {
      background-color: none !important;
      opacity: 0.7;
      color: #9d9d9d;
    }
    &:focus {
      background-color: none !important;
    }
    &.active {
      color: #9d9d9d !important;
    }
  }
  & button {
    display: block;
    cursor: pointer;
    background: none;
    color: #9d9d9d;
    text-align: center;
    width: 2rem;
    min-width: 0;
    height: 3rem;
    // padding: 20px;
    margin: 1rem;
    border: none;
    align-self: flex-start;
    &:hover {
      background-color: none !important;
    }
    &:focus {
      background-color: none;
      border: none;
      -moz-box-shadow: none !important;
      -webkit-box-shadow: none !important;
      box-shadow: none !important;
      outline: 0;
    }
    &:active {
      color: black;
      background: #eee;
      text-decoration: underline;
    }
  }
}

@media screen and (min-width: 40rem) {
  nav {
    & ul {
      width: 100%;
      & li,
      &.hidden li:not(:first-child),
      &.hidden li:first-child {
        display: inline;
      }
      & li:nth-of-type(1n+3) {
        float: right;
      }
    }
    & button {
      display: none;
    }
  }
}
