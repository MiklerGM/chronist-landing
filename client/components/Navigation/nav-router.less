@import "../../colors.less";

/* Navbar BEGIN */
#Navigation {
  background: linear-gradient(rgba(0, 0, 0, 0.1), rgba(0, 0, 0, 0.1)), url("../../images/back-progressive.jpg");
  background-color: transparent;
  background-repeat: no-repeat;
  // parallax thing
  background-attachment: fixed;
  background-position: center;
  background-size: cover;
  & a:focus { background-color: none !important; }
}

ul.topnav {
  // position: absolute;
  // top: 0;

  list-style-type: none;
  overflow: hidden;
  // dbackground: none;
  padding: 20px;

  & button {
    background: none;
    color: #9d9d9d;
    text-align: center;
    padding: 20px;
    border: none;

    &:hover { background-color: none !important; }

    &:focus {
      background-color: none;
      border: none;
      // color:red;
      -moz-box-shadow: none !important;
      -webkit-box-shadow: none !important;
      box-shadow: none !important;
      outline: 0;
    }

    &:active {
      color: black;
      background: #eee;
      text-decoration: underline;
    }
  }

  & li {
    float: left;
    // width: 60%;
    & a {
      display: inline-block;
      color: #9d9d9d;
      text-align: center;
      padding: 20px;
      text-decoration: none;
      transition: 0.3s;
      &:hover { background-color: none !important; }
    }

    &.icon { display: none; }
  }
}

@media screen and (max-width: 600px) {
  ul.topnav li {
    &:not(:first-child) { display: none; }

    &.icon {
      float: right;
      display: inline-block;
      position: absolute;
      top: 0;
      right: 0;
      padding-top: 20px;
      padding-right: 55px;
    }
  }
}

@media screen and (max-width: 600px) {
  ul.topnav.responsive {
    position: relative;

    & li {
      float: none;
      display: inline;

      &.icon {
        float: right;
        display: inline-block;
        position: absolute;
        top: 0;
        right: 0;
        padding-top: 20px;
        padding-right: 40px;
      }

      & a {
        display: block;
        width: 60%;
        text-align: left;
      }
    }
  }
}
