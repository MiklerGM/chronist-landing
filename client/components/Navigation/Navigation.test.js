import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import IntlWrapper from '../../IntlWrapper';
import Navigation from './Navigation';

const TestComponent = () => (
  <IntlWrapper>
    <Router>
      <Navigation
        locale='ru'
        onChangeLanguage={(e) => console.log('dumb func', e)}
      />
    </Router>
  </IntlWrapper>
);

// it must render
it('Navigation render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

it('Navigation render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

// Snapshot tests
it('Navigation Snapshot', () => {
  const tree = renderer
    .create(<TestComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
