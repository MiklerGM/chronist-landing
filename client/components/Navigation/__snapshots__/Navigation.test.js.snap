// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Snapshot 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <nav>
      <ul
        className=""
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            <span>
              Хронист
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://chronmaps.com/"
          >
            <span>
              Карта
            </span>
          </a>
        </li>
        <li>
          <a
            onClick={[Function]}
          >
            EN
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/blog"
            onClick={[Function]}
          >
            <span>
              Блог
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://vk.com/chronist"
          >
             VK 
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/faq"
            onClick={[Function]}
          >
            FAQ
          </a>
        </li>
      </ul>
      <button
        className="icon"
        onClick={[Function]}
      >
        <i
          className="fa icon-menu"
        />
      </button>
    </nav>
  </div>
</div>
`;
